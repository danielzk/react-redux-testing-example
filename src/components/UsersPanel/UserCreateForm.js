import React from 'react'

export default class UserCreateForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      firstName: '',
      lastName: '',
    }
  }

  onUsernameChange = (event) => {
    this.setState({username: event.target.value})
  }

  onFirstNameChange = (event) => {
    this.setState({firstName: event.target.value})
  }

  onLastNameChange = (event) => {
    this.setState({lastName: event.target.value})
  }

  onSubmit = (event) => {
    this.props.actions.createUser(
      this.state.username, this.state.firstName, this.state.lastName)
    event.preventDefault()
  }

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input
          id="username"
          type="text"
          value={this.state.username}
          onChange={this.onUsernameChange}
          placeholder="Username"
        />

        <input
          id="first-name"
          type="text"
          value={this.state.firstName}
          onChange={this.onFirstNameChange}
          placeholder="First name"
        />

        <input
          id="last-name"
          type="text"
          value={this.state.lastName}
          onChange={this.onLastNameChange}
          placeholder="Last name"
        />

        <button type="submit">Create</button>
      </form>
    )
  }
}
