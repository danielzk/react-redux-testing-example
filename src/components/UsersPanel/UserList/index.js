import React from 'react'

import UserItem from './UserItem'

export default class UserList extends React.Component {
  render() {
    return (
      <div>
        {this.props.users.count() > 0 &&
          <ul>
            {this.props.users.toArray().map(user =>
              <UserItem user={user} key={user.id} {...this.props}/>
            )}
          </ul>
        }

        {this.props.users.count() == 0 &&
          <span>No users</span>
        }
      </div>
    )
  }
}
