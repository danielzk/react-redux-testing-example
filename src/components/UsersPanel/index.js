import React from 'react'

import UserCreateForm from './UserCreateForm'
import UserList from './UserList'

export default class UsersPanel extends React.Component {
  constructor(props) {
    super(props)
    this.props.actions.loadUsers()
  }

  render() {
    return (
      <div>
        <UserCreateForm {...this.props}/>
        <UserList {...this.props}/>
      </div>
    )
  }
}
