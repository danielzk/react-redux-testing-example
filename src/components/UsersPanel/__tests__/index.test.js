import React from 'react'
import {shallow} from 'enzyme'

import UsersPanel from '../'

describe('UsersPanel component', () => {
  it('should render UserCreateForm and UserList, and call loadUsers', () => {
    const props = {
      actions: {
        loadUsers: jest.fn(),
      }
    }
    const wrapper = shallow(<UsersPanel {...props}/>)
    expect(wrapper).toMatchSnapshot()
    expect(props.actions.loadUsers).toHaveBeenCalledTimes(1)
  })
})
