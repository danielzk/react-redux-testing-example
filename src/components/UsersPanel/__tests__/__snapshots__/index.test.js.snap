// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPanel component should render UserCreateForm and UserList, and call loadUsers 1`] = `
<div>
  <UserCreateForm
    actions={
      Object {
        "loadUsers": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
      }
    }
  />
  <UserList
    actions={
      Object {
        "loadUsers": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
      }
    }
  />
</div>
`;
